src/include/processor/result/result_set.h

```diff
@@ -17,6 +17,9 @@ class ResultSet {
         dataChunks[pos] = std::move(dataChunk);
     }
 
+    inline std::shared_ptr<common::DataChunk> getDataChunk(data_chunk_pos_t dataChunkPos) {
```

Comment:
I think it is better to return a raw pointer here, also the caller calls `get()` right after `getDataChunk`:
```
 aggregateInput->multiplicityChunks.push_back(
                resultSet->getDataChunk(dataChunkPos).get());
```